lib/recursive-methods.rb

```diff
@@ -1,49 +1,90 @@
 # Authoring recursive algorithms. Add comments including time and space complexity for each method.
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) where n is the size of the number input
+# Space complexity: O(n) where n is the size of the number input
 def factorial(n)
-    raise NotImplementedError, "Method not implemented"
+  if n < 0
+    raise ArgumentError
+  elsif n == 0 || n == 1
+    return 1
+  else
+    return n * (factorial(n - 1))
+  end
 end
 
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n) where n is the length of the string
```

Comment:
This is actually O(n^2) for time and space due to the fact that `s[1..-1]` creates a **new** string with each recursive call.